frappe/tests/test_client.py

```diff
@@ -141,3 +141,41 @@ def test_client_get(self):
 		self.assertEqual(get("ToDo", filters=filters_json).description, "test")
 
 		todo.delete()
+
+	def test_client_insert(self):
+		from frappe.client import insert
+
+		def get_random_title():
+			return "test-{0}".format(frappe.generate_hash())
+
+		# test insert dict
+		doc = {"doctype": "Note", "title": get_random_title(), "content": "test"}
+		note1 = insert(doc)
+		self.assertTrue(note1)
+
+		# test insert json
+		doc["title"] = get_random_title()
+		json_doc = frappe.as_json(doc)
+		note2 = insert(json_doc)
+		self.assertTrue(note2)
+
+		# test insert child doc without parent fields
+		child_doc = {"doctype": "Note Seen By", "user": "Administrator"}
+		with self.assertRaises(frappe.ValidationError):
+			insert(child_doc)
+
+		# test insert child doc with parent fields
+		child_doc = {
+			"doctype": "Note Seen By",
+			"user": "Administrator",
+			"parenttype": "Note",
+			"parent": note1.name,
+			"parentfield": "seen_by",
+		}
+		note3 = insert(child_doc)
+		self.assertTrue(note3)
+
+		# cleanup
+		frappe.delete_doc("Note", note1.name)
+		frappe.delete_doc("Note", note2.name)
+		frappe.delete_doc("Note", note3.name)
```

Comment:
this line is not needed ? 
`note3` will have the same name as `note1` as we return the whole parent doc (with child doc appended)